src/operations/write.ts

```diff
@@ -61,6 +63,56 @@ export const write = (
   return result;
 };
 
+export const writeFragment = (
+  store: Store,
+  query: DocumentNode,
+  data: Data
+) => {
+  if (process.env.NODE_ENV !== 'production' && query.definitions.length > 1) {
+    throw new Error(
+      'You can only pass one fragment when writing to a fragment.'
+    );
+  }
+
+  (query.definitions as FragmentDefinitionNode[]).forEach(
```

Comment:
The iteration seems a little obsolete due to the invariant above. Shall we check for both length and kind above and then just grab the first node?